public/app/plugins/datasource/tempo/traceql/TempoQueryBuilderOptions.tsx

```diff
@@ -26,7 +26,7 @@ export const TempoQueryBuilderOptions = React.memo<Props>(({ onChange, query })
               placeholder="auto"
               type="number"
               min={1}
-              defaultValue={DEFAULT_LIMIT}
+              defaultValue={query.limit || DEFAULT_LIMIT}
```

Comment:
@joey-grafana @CrypticSignal I don't think this is the right fix here. The `defaultValue` I would assume should not be shown if `query.limit` is set. If it is set there should be no need to override the defaultValue prop. If it isn't set this should not have any effect. Seems like there is some other root cause why this does not work as expected.

So either we are using defaultValue wrong here, it does not work as expected and there is a bug in the Input or there is something else going on.

Reply:
Also if query.limit === 0 this will show DEFAULT_LIMIT. Not sure if that is expected or not.

Reply:
Are you saying that we shouldn't have the `defaultValue` prop at all? If it think about it, I don't think we need it because if we have `value={DEFAULT_LIMIT}`, that is essentially setting a default value, so I'm not entirely sure of the purpose of the `defaultValue` prop.

I will do some testing to check whether query limiting works as intended. If not, I will make the necessary changes in a new PR.